include/deal.II/base/partitioner.h

```diff
@@ -58,7 +59,7 @@ namespace Utilities
      *
      * @author Katharina Kormann, Martin Kronbichler, 2010, 2011
      */
-    class Partitioner
+    class Partitioner : public ::dealii::LinearAlgebra::CommunicationPatternBase
```

Comment:
```CommunicationPatternBase``` was made very small not by choice but because I wasn't sure what would be common between all the vectors. If you think some of the functions in ```Partitioner``` could be useful to the others vectors just move it to the base class.